assets/js/components/settings/SettingsActiveModule/Header.js

```diff
@@ -184,6 +183,45 @@ export default function Header( { slug } ) {
 		return null;
 	}
 
+	// Do not show a "Connected" status for the Analytics module if GA4 is not connected.
+	const showAsConnected =
+		connected && ( 'analytics' !== slug || isGA4Connected );
+
+	let moduleStatus = null;
+
+	if ( showAsConnected ) {
+		moduleStatus = <p>{ __( 'Connected', 'google-site-kit' ) }</p>;
+	} else if ( 'analytics' === slug && connected && ! isGA4Connected ) {
+		if ( hasAnalyticsAccess ) {
+			moduleStatus = (
+				<Button onClick={ handleConnectGA4ButtonClick }>
+					{ __( 'Connect Google Analytics 4', 'google-site-kit' ) }
+				</Button>
+			);
+		} else if ( hasResolvedAnalyticsAccess ) {
+			moduleStatus = (
+				<p>
+					{ __(
+						'Google Analytics 4 is not connected',
+						'google-site-kit'
+					) }
+				</p>
+			);
+		} else {
+			moduleStatus = <ProgressBar height={ 32 } small />;
```

Comment:
I noticed there is a visual shift between loading and loaded states. I'm not sure if this is entirely due to the height of the progress bar or not but we should aim to keep the height of the header consistent for all cases (loading, button, or text status).

Reply:
Updated all cases to have an equal height, thank you for the call!